pkg/provider/gcp/secretmanager/workload_identity.go

```diff
@@ -78,11 +78,11 @@ type saTokenGenerator interface {
 }
 
 func newWorkloadIdentity(ctx context.Context, projectID string) (*workloadIdentity, error) {
-	iamc, err := newIAMClient(ctx)
+	satg, err := newSATokenGenerator()
 	if err != nil {
 		return nil, err
 	}
-	satg, err := newSATokenGenerator()
+	iamc, err := newIAMClient(ctx)
```

Comment:
The following change also works, but I took the approach that changes the order of `newSATokenGenerator` and `newIAMClient` for simplicity.
```golang
	iamc, err := newIAMClient(ctx)
	if err != nil {
		return nil, err
	}
	satg, err := newSATokenGenerator()
	if err != nil {
		iamc.Close()
		return nil, err
	}
```